### PR TITLE
Add public primitive and semantic `shadow` token scale

### LINKED ISSUE #10511: Add public primitive `shadow` token layer

### LINKED ISSUE #10512: Update semantic `shadow` token layer

### PR DESCRIPTION
**Pull request by lgriffee**
### WHY are these changes introduced?

Fixes #10511
Fixes #10512

### WHAT is this pull request doing?

Adds the following values to new public primitive and semantic `shadow` token scales: 

| New Token          | Value        |
| ------------------------- | ------------------------ |  
| `--p-shadow-0` | `none`    |
| `--p-shadow-100` | `0px 0px 2px rgba(31, 33, 36, 0.24)`    |
| `--p-shadow-200` | `0px 1px 1px rgba(31, 33, 36, 0.1)`    |
| `--p-shadow-300` | `0px 2px 4px rgba(31, 33, 36, 0.1), 0px 1px 6px rgba(31, 33, 36, 0.05)`    |
| `--p-shadow-400` | `0px 4px 12px rgba(31, 33, 36, 0.2), 0px 2px 6px rgba(31, 33, 36, 0.05)`    |
| `--p-shadow-500` | `0px 4px 18px -2px rgba(31, 33, 36, 0.08), 0px 12px 18px -2px rgba(31, 33, 36, 0.15)`    |
| `--p-shadow-600` | `0px 32px 32px rgba(31, 33, 36, 0.15), 0px 32px 56px -2px rgba(31, 33, 36, 0.16)`    |
| `--p-shadow-bevel-100` | `1px 0px 0px 0px rgba(0, 0, 0, 0.13) inset, -1px 0px 0px 0px rgba(0, 0, 0, 0.13) inset, 0px -1px 0px 0px rgba(0, 0, 0, 0.17) inset, 0px 1px 0px 0px rgba(204, 204, 204, 0.5) inset`    |
| `--p-shadow-inset-100` | `inset 0px 0px 3px rgba(31, 33, 36, 0.56)`    |
| `--p-shadow-inset-200` | `inset 0px 2px 4px rgba(31, 33, 36, 0.32)`    |
| `--p-shadow-button` | `inset 0px -1px 0px #B5B5B5, inset -1px 0px 0px #E3E3E3, inset 1px 0px 0px #E3E3E3, inset 0px 1px 0px #E3E3E3`    |
| `--p-shadow-button-hover` | `inset 0px -1px 0px #CCCCCC, inset 1px 0px 0px #EBEBEB, inset -1px 0px 0px #EBEBEB, inset 0px 1px 0px #EBEBEB`    |
| `--p-shadow-button-inset` | `--p-shadow-inset-200`    |
| `--p-shadow-button-primary` | `0px 2px 0px 0px rgba(255, 255, 255, 0.2) inset, 2px 0px 0px 0px rgba(255, 255, 255, 0.2) inset, -2px 0px 0px 0px rgba(255, 255, 255, 0.2) inset, 0px -1px 0px 1px #000 inset, 0px 1px 0px 0px #000 inset`    |
| `--p-shadow-button-primary-inset` | `0px 3px 0px 0px #000 inset`    |
| `--p-shadow-button-primary-hover` | `0px 1px 0px 0px rgba(255, 255, 255, 0.24) inset, 1px 0px 0px 0px rgba(255, 255, 255, 0.16) inset, -1px 0px 0px 0px rgba(255, 255, 255, 0.16) inset, 0px -1.5px 0px 0px rgba(255, 255, 255, 0.07) inset, 0px 0px 0px 0.5px #1A1A1A`    |
| `--p-shadow-button-primary-critical` | `0px 1px 0px 0px rgba(255, 255, 255, 0.4) inset, 1px 0px 0px 0px rgba(255, 255, 255, 0.2) inset, -1px 0px 0px 0px rgba(255, 255, 255, 0.2) inset, 0px -1.5px 0px 0px rgba(0, 0, 0, 0.25) inset`    |
| `--p-shadow-button-primary-critical-hover` | `-1px 0px 0px 0px rgba(255, 255, 255, 0.20) inset, 1px 0px 0px 0px rgba(255, 255, 255, 0.20) inset, 0px 0.5px 0px 0px rgba(0, 0, 0, 0.25), 0px -1.5px 0px 0px rgba(255, 255, 255, 0.07) inset, 0px 1px 0px 0px rgba(255, 255, 255, 0.24) inset`    |
| `--p-shadow-button-primary-critical-inset` | `0px 2px 0px 0px rgba(0, 0, 0, 0.60) inset, 1px 0px 1px 0px rgba(0, 0, 0, 0.20) inset, -1px 0px 1px 0px rgba(0, 0, 0, 0.20) inset`    |
| `--p-shadow-button-primary-success` | `0px 1px 0px 0px rgba(255, 255, 255, 0.4) inset, 1px 0px 0px 0px rgba(255, 255, 255, 0.2) inset, -1px 0px 0px 0px rgba(255, 255, 255, 0.2) inset, 0px -1.5px 0px 0px rgba(0, 0, 0, 0.25) inset`    |
| `--p-shadow-button-primary-success-hover` | `-1px 0px 0px 0px rgba(255, 255, 255, 0.20) inset, 1px 0px 0px 0px rgba(255, 255, 255, 0.20) inset, 0px 0.5px 0px 0px rgba(0, 0, 0, 0.25), 0px -1.5px 0px 0px rgba(255, 255, 255, 0.07) inset, 0px 1px 0px 0px rgba(255, 255, 255, 0.24) inset`    |
| `--p-shadow-button-primary-success-inset` | `0px 2px 0px 0px rgba(0, 0, 0, 0.60) inset, 1px 0px 1px 0px rgba(0, 0, 0, 0.20) inset, -1px 0px 1px 0px rgba(0, 0, 0, 0.20) inset`    |
| `--p-shadow-border-inset` | `0px 0px 0px 1px rgba(0, 0, 0, 0.08) inset`    |

### CODE REVIEW

File: .changeset/light-cherries-compare.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+'@shopify/polaris-tokens': minor
+---
+
+Added public primitive and semantic `shadow` token scale
```
Comment (lgriffee) [inline .changeset/light-cherries-compare.md]:
```suggestion
Added public primitive and semantic `shadow` token scales
```

**Review by aaronccasanova — APPROVED** [review]

**Review by bernardojoaogarcia — APPROVED** [review]
Looks great! 🚀 @lgriffee Are you going to be adding a new `shadow-button-inset` token which can be mapped to `shadow-inset-200` ?

**Comment by lgriffee** [comment]
> Looks great! 🚀 @lgriffee Are you going to be adding a new `shadow-button-inset` token which can be mapped to `shadow-inset-200` ?

@bernardojoaogarcia Good catch I missed that one! I've added the following in 67ccfe9 to this PR: 

| New Token         | Value         |
| ------------------------- | ------------------------ |
| `--p-shadow-button-inset` | `--p-shadow-inset-200`    |

~Quick question with regards the the migration of `--p-shadow-inset-md` it should be mapped to `--p-shadow-button-inset` in cases where it is used inside a button but to  `--p-shadow-inset-200` in all other cases? If so, I'll just need to make sure I make a note that this migration needs to be done manually to account for that.~ (see comment below)

| Deprecated CSS Custom Property            | Replacement Value                  |
| ----------------------------------------- | ---------------------------------- |
| `--p-shadow-inset-md` | `--p-shadow-inset-200`<br>`--p-shadow-button-inset`    |

**Comment by lgriffee** [comment]
> Quick question with regards the the migration of --p-shadow-inset-md it should be mapped to --p-shadow-button-inset in cases where it is used inside a button but to --p-shadow-inset-200 in all other cases? If so, I'll just need to make sure I make a note that this migration needs to be done manually to account for that.

Actually I chatted with @aaronccasanova about this and it's safer if we just migrate `--p-shadow-inset-md` to only `--p-shadow-inset-200` for now and handle applying the semantic token incrementally over time.

| Deprecated CSS Custom Property            | Replacement Value                  |
| ----------------------------------------- | ---------------------------------- |
| `--p-shadow-inset-md` | `--p-shadow-inset-200`|

**Pull request merged**